### PR TITLE
Allow for date range selection

### PR DESCRIPTION
**Pull request by xzyfer**
This patch is a fork of #79 with the suggested modifications and
merge conflicts resolved. The credit should go to the original
author. This is simply something I wanted and the original author
appears to have gone MIA.

### CODE REVIEW

**Comment by xzyfer** [comment]
Refactored and added test coverage for the `inRange` detection.

**Comment by RSO** [comment]
:+1: Awesome stuff, thanks!

File: src/day.jsx [hunk 1]
```diff
@@ -22,12 +22,14 @@ var Day = React.createClass( {
     if ( this.props.day.sameDay( this.props.selected ) )
       classes.push( "datepicker__day--selected" );
 
+    if ( this.props.day.inRange )
```
Comment (rafeememon) [inline src/day.jsx]: I believe this should be `this.props.inRange`; `day.inRange` is a method that will always be defined/truthy

Comment (xzyfer) [inline src/day.jsx]: You could be right. I mostly just rebased the existing PR. Without the
example page I don't have an environment to easily run this component so
I'm relying on the tests.

IMHO bringing back the example page would really help with contributions.

Comment (rafeememon) [inline src/day.jsx]: Hey @xzyfer, no prob -- I too was finding myself stumbling to find the right way to bootstrap this.  I'm working on a PR that will begin a test suite for the Day component.

**Comment by martijnrusschen** [comment]
Having some issues with creating an example for this awesome thing. Mind looking into it what's going wrong? https://github.com/Hacker0x01/react-datepicker/pull/219

**Comment by xzyfer** [comment]
Sorry, timezones. It's late here I'll happily take a look in the morning.

**Pull request merged**